geoh5py/shared/conversion/grid2d_to_geoimage.py

```diff
@@ -0,0 +1,172 @@
+        """
+        Convert a data of the Grid2D to a numpy array
+        with a format compatible with :obj:`PIL.Image` object.
+        :param data_name: the name of the data to extract.
+        :return: the data formatted with the right shape,
+        beetweem 0 and 255, as uint8.
+        """
+        # get the data
+        data = self.entity.get_data(data_name)[0].values
+
+        # reshape them
+        data = data.reshape(self.entity.v_count, self.entity.u_count)[::-1]
+
+        # normalize them #todo: change the no number value in np.nan
+        min_, max_ = np.nanmin(data), np.nanmax(data)
+        data = (data - min_) / (max_ - min_)
+        data *= 255
+
+        return data.astype(np.uint8)
+
+    def __call__(self, data_list: list | str, **geoimage_kwargs) -> GeoImage:
+        """
+        Create a :obj:geoh5py.objects.geo_image.GeoImage object from the current Grid2D.
+        :param data_list: the list of the data name to pass as band in the image.
+        The len of the list can only be 1, 3, 4.
+        :param **geoimage_kwargs: any argument of :obj:`geoh5py.objects.geo_image.GeoImage`.
+        :return: a new georeferenced :obj:`geoh5py.objects.geo_image.GeoImage`.
+        """
+        # catch exception if str is entered instead of a list
+        if isinstance(data_list, str):
+            data_list = [data_list]
+
+        # prepare the image
+        if isinstance(data_list, list):
+            if len(data_list) not in [1, 3, 4]:
+                raise IndexError("Only 1, 3, or 4 layers can be selected")
+
+            data = np.empty((self.entity.v_count, self.entity.u_count, 0)).astype(
+                np.uint8
+            )
+            for data_name in data_list:
+                if data_name not in self.entity.get_data_list():
+                    raise KeyError(
+                        f"The key {data_list} is not in the data: {self.entity.get_data_list()}"
+                    )
+
+                data_temp = self.data_to_pil_format(data_name)
+                data = np.dstack((data, data_temp))
+        else:
+            raise TypeError(
+                "The type of the keys must be a string or a list,",
+                f"but you entered an {type(data_list)} ",
+            )
+
+        # convert to PIL
+        if data.shape[-1] == 1:
+            image = Image.fromarray(data[:, :, 0], mode="L")
+        elif data.shape[-1] == 3:
+            image = Image.fromarray(data, mode="RGB")
+        elif data.shape[-1] == 4:
+            image = Image.fromarray(data, mode="CMYK")
+
+        workspace = geoimage_kwargs.get("workspace", self.entity.workspace)
+        geoimage_kwargs.pop("workspace", None)
+
+        # create a geoimage
+        new_geoimage = objects.GeoImage.create(
+            workspace, image=image, tag=self.grid_to_tag(), **geoimage_kwargs
+        )
+
+        # georeference it
+        new_geoimage.georeferencing_from_tiff()
```

Comment:
Same here, we could call two methods. The first one translates the attribute of one object to the other. Second method (optional to the call) transfer the data.